mmdet/datasets/coco.py

```diff
@@ -384,18 +386,19 @@ def evaluate(self,
                 print_log(log_msg, logger=logger)
                 continue
 
-            if metric not in result_files:
-                raise KeyError(f'{metric} is not in results')
+            metric_type = 'bbox' if metric == 'f1' else metric
+            if metric_type not in result_files:
+                raise KeyError(f'{metric_type} is not in results')
             try:
-                cocoDt = cocoGt.loadRes(result_files[metric])
+                cocoDt = cocoGt.loadRes(result_files[metric_type])
             except IndexError:
                 print_log(
                     'The testing results of the whole dataset is empty.',
                     logger=logger,
                     level=logging.ERROR)
                 break
 
-            iou_type = 'bbox' if metric == 'proposal' else metric
+            iou_type = 'bbox' if metric == 'proposal' or 'f1' else metric
```

Comment:
```suggestion
            iou_type = 'bbox' if metric in {'proposal', 'f1'} else metric
```

Reply:
Is there some secret reason why do you prefer dictionary instead  of list?

Reply:
It's a set, not a dict. Sets and list has different complexity of element search operation. No difference here, though.